### PR TITLE
Submit PNP models for ADU PPR

### PR DESCRIPTION
**Pull request by xinanqiu**
New PNP models for ADU PPR scenario

### Thank you for contributing to the Azure IoT Plug and Play Models repository

This project has adopted the [Microsoft Open Source Code of Conduct](https://opensource.microsoft.com/codeofconduct/). For more information see the [Code of Conduct FAQ](https://opensource.microsoft.com/codeofconduct/faq/) or contact [opencode@microsoft.com](mailto:opencode@microsoft.com) with any additional questions or comments.

The repository pipeline will ensure minimum validation of incoming DTDL models.  

- PR validation steps are described in the tools [Wiki](https://github.com/Azure/iot-plugandplay-models-tools/wiki/Validation-Pipeline#pr-validation-checks).
- The same validation can be run locally via the [dmr-client](https://github.com/Azure/iot-plugandplay-models-tools/tree/dev/clients/dotnet/Microsoft.IoT.ModelsRepository.CommandLine) command line tool.

## Requested info template for model(s) submission

When submitting models to the repository we ask that you provide as much of the following meta information around your models and related devices as possible. This info will be used to improve Plug and Play.

:star2: Please replace the markdown comment examples with your own values.

### CODE REVIEW

**Review by rido-min — APPROVED** [review]

**Review by jrdouceur — DISMISSED** [review]
LGTM

File: dtmi/azure/iot/deviceupdate-1.json [hunk 1]
```diff
@@ -0,0 +1,163 @@
+{
+  "@id": "dtmi:azure:iot:deviceUpdate;1",
+  "@type": "Interface",
+  "description": "Device Update Interface",
```
Comment (briancr-ms) [inline dtmi/azure/iot/deviceupdate-1.json:4]: Should the description format follow the guidelines? https://github.com/Azure/iot-plugandplay-models-tools/wiki/DTDL-Naming-Guidelines#interfaces

File: dtmi/azure/iot/deviceupdatecontractmodel-1.json [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+  "@type": "Interface",
+  "displayName": "Device Update Model V1",
+  "description": "The model supports update manifest version 4",
+  "contents": [
+    {
+      "@type": "Component",
+      "schema": "dtmi:azure:iot:deviceUpdate;1",
+      "name": "deviceUpdate",
+      "displayName": "Device Update Plug and Play Component",
+      "description": "Device Update Plug and Play Component"
```
Comment (briancr-ms) [inline dtmi/azure/iot/deviceupdatecontractmodel-1.json:12]: Should the description follow the guidelines? https://github.com/Azure/iot-plugandplay-models-tools/wiki/DTDL-Naming-Guidelines

File: dtmi/azure/iot/diagnosticinformation-1.json [hunk 3]
```diff
@@ -0,0 +1,54 @@
+{
+  "@id": "dtmi:azure:iot:diagnosticInformation;1",
+  "@type": "Interface",
+  "description": "Diagnostic Information Interface",
```
Comment (briancr-ms) [inline dtmi/azure/iot/diagnosticinformation-1.json:4]: Should the description follow the guidelines? https://github.com/Azure/iot-plugandplay-models-tools/wiki/DTDL-Naming-Guidelines#interfaces

Comment (xinanqiu) [reply dtmi/azure/iot/diagnosticinformation-1.json]: Fixed description

File: dtmi/azure/iot/deviceupdate-1.json (hunk 1 above)
Comment (xinanqiu) [reply dtmi/azure/iot/deviceupdate-1.json]: Fixed description

File: dtmi/azure/iot/deviceupdatecontractmodel-1.json [hunk 4]
```diff
@@ -0,0 +1,16 @@
+{
+  "@id": "dtmi:azure:iot:deviceUpdateContractModel;1",
+  "@type": "Interface",
+  "displayName": "Device Update Model V1",
+  "description": "The model supports update manifest version 4",
```
[the remark this answers is not in the document]
Comment (xinanqiu) [reply dtmi/azure/iot/deviceupdatecontractmodel-1.json]: Fixed description

File: dtmi/azure/iot/deviceupdatecontractmodel-1.json (hunk 2 above)
Comment (xinanqiu) [reply dtmi/azure/iot/deviceupdatecontractmodel-1.json:12]: Fixed description

File: dtmi/azure/iot/deviceupdatemodel-1.json [hunk 5]
```diff
@@ -0,0 +1,24 @@
+{
+  "@id": "dtmi:azure:iot:deviceUpdateModel;1",
+  "@type": "Interface",
+  "displayName": "Device Update Model",
+  "description": "Device Update Model",
```
[the remark this answers is not in the document]
Comment (xinanqiu) [reply dtmi/azure/iot/deviceupdatemodel-1.json]: Fixed description

File: dtmi/azure/iot/diagnosticinformation-1.json (hunk 3 above)
Comment (xinanqiu) [reply dtmi/azure/iot/diagnosticinformation-1.json]: Fixed description

File: dtmi/azure/iot/deviceupdate-1.json [hunk 6]
```diff
@@ -0,0 +1,163 @@
...
+          "schema": "integer"
+        },
+        {
+          "name": "resultDetails",
+          "schema": "string"
+        }
+      ]
+    }
+  ],
+  "@context": "dtmi:dtdl:context;2"
```
Comment (rido-min) [inline dtmi/azure/iot/deviceupdate-1.json:162]: I'd suggest to move the `@context` to the first line in the interface

Comment (xinanqiu) [reply dtmi/azure/iot/deviceupdate-1.json]: Fixed it

**Review by rido-min — APPROVED** [review]
LGTM

**Review by briancr-ms — APPROVED** [review]
Looks good, thanks!

**Pull request merged**